frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -154,20 +164,34 @@ class Gallery extends React.Component {
     // app crash telling it cannot render a Link outside a router
     // see https://github.com/tracim/tracim/issues/1637
     // GLOBAL_dispatchEvent({type: 'setBreadcrumbs', data: {breadcrumbs: breadcrumbsList}})
-    this.setState({ breadcrumbsList: breadcrumbsList })
+    return breadcrumbsList
   }
 
   loadFolderDetail = async (workspaceId, folderId) => {
     const { state, props } = this
 
-    const fetchContentDetail = await handleFetchResult(
+    let fetchContentDetail = await handleFetchResult(
       await getFolderDetail(state.config.apiUrl, workspaceId, folderId)
     )
 
     switch (fetchContentDetail.apiResponse.status) {
       case 200:
-        this.setState({ fileName: fetchContentDetail.body.filename })
-        break
+        const fileName = fetchContentDetail.body.filename
+        let folderParentsId = fetchContentDetail.body.parent_id ? [fetchContentDetail.body.parent_id] : []
+
+        while (fetchContentDetail.body.parent_id !== null) {
+          fetchContentDetail = await handleFetchResult(
+            await getFolderDetail(state.config.apiUrl, workspaceId, fetchContentDetail.body.parent_id)
+          )
+          if (fetchContentDetail.apiResponse.status === 200) {
+            folderParentsId.push(fetchContentDetail.body.parent_id)
+          } else {
+            this.sendGlobalFlashMessage(props.t('Error while loading folder detail'))
+            break
+          }
+        }
+
+        return { fileName, folderParentsId }
```

Comment:
if your function return an object, it must always return something predetermined. In some cases, your function will return undefined. This is highly error prone